### ISSUE TITLE
Velero 1.7.1 rc.1: Connection to minio failing when credential field is not explicitly specified in BSL

### ISSUE THREAD

**Issue by lautenbde** [vmware-tanzu/velero#4395]
**What steps did you take and what happened:**
Installed Velero 1.7.1 rc (from scratch, uninstalled 1.7.0 first).
Created BSL "default" as usual configured to connect to local minio with self-signed certificate specified in caCert field.
Created default secret "cloud-credentials" with key "cloud" holding the minio credentials as usual
Tried to create backup which is failing. Also sync or any other connection to minio is failing as apparent from velero logs:

"
time="2021-11-24T14:43:46Z" level=error msg="Error getting backup store for this location" backupLocation=default controller=backup-sync error="rpc error: code = Unknown desc = NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors" error.file="/go/src/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:59" error.function=main.getSession logSource="pkg/controller/backup_sync_controller.go:175"

I then added credentials field to the BSL:

  credential:
    key: cloud
    name: cloud-credentials

After this, all back to normal, connection to minio and backup successful.

Could this issue be related to this change: 

https://github.com/vmware-tanzu/velero/commit/e4019f26c1ab215c4dbe0e2fa5eb8b3101d298bf

**What did you expect to happen:**

Backups can be created successfully without adding the default credential secret to BSL explicitly.

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): 1.7.1 rc.1
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`): 1.18
- Kubernetes installer & version:
- Cloud provider or hardware configuration: local minio 
- OS (e.g. from `/etc/os-release`): RHEL 7.9

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by reasonerjt** [comment]
I don't think this commit is cherry picked to v1.7 branch 
@ywk253100  could you confirm?

**Comment by reasonerjt** [comment]
@lautenbde please provide the debug bundle as requested in the issue template.

**Comment by lautenbde** [comment]
Adding logs resulting from velero debug --backup command. The first shows backup failing as Credential field was not specified in BSL explicitly:

[bundle-2021-11-24-17-16-03.tar.gz](https://github.com/vmware-tanzu/velero/files/7597248/bundle-2021-11-24-17-16-03.tar.gz)

This one shows successful backup after adding Credential field to BSL: 

[bundle-2021-11-24-17-18-57.tar.gz](https://github.com/vmware-tanzu/velero/files/7597246/bundle-2021-11-24-17-18-57.tar.gz)

Note: velero client can't access minio due to how networking is setup so the actual backup logs are missing. However, in case of connection to minio is failing there won't be any anyway.

**Comment by ywk253100** [comment]
No, the commit isn't included in the 1.7 branch

**Comment by lautenbde** [comment]
@reasonerjt Are the logs above helpful? Please let me know if you need additional information. Thanks!

**Comment by reasonerjt** [comment]
@lautenbde  
I checked the log, I noticed in the output of `backup_describe_test9.txt` and `backup_test9.log`  in the bundle `backup_test10.log`

There's error message 
```
error getting backup resource list: Get "https://10.107.140.118:9000/velero/backups/test9/test9-resource-list.json.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=dummyminiouser%2F20211124%2Fminio%2Fs3%2Faws4_request&X-Amz-Date=20211124T161605Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=045dfd70fa885f8b872206dfbe6f30225010ef7d7281bfd70fd8cc98d142ca77": x509: cannot validate certificate for 10.107.140.118 because it doesn't contain any IP SANs
```

So the certificate of your minio instance is incorrect, it has to contain IP SANs to be trusted by the client, otherwise the log which is stored in the object store can't be collected.

As for the error you see, per the doc:
https://velero.io/docs/v1.7/api-types/backupstoragelocation/

The field `caCert` is only for tls connection, you still need to set the credentials, such as Access key and Secret key in the BSL.

**Comment by lautenbde** [comment]
@reasonerjt Yes, as mentioned in my original comment I am aware of the fact that log retrieval from minio is failing due to IP address. However, the credentials used to be and I think still are optional. By default, the credentials are implicitly fetched from secret "cloud-credentials" / key "cloud" as mentioned in my original comment. It was always working this way up to and including 1.7.0 and now with 1.7.1 rc.1 exactly the same BSL in combination as specified before is no more working. So something has changed. 

I now have fetched the test10 log file from minio directly:
[test10-logs.gz](https://github.com/vmware-tanzu/velero/files/7614344/test10-logs.gz)

Obviously, no test9 related folder/files located on minio due to failed connection.

**Comment by reasonerjt** [comment]
@lautenbde  thanks for the reply,
By default, if not set in the BSL, the credentials are fetched by aws sdk from the env var `AWS_SHARED_CREDENTIALS_FILE` which is mounted via the secret `cloud-credentials` 

I checked the commits added between v1.7.0 and v1.7.1-rc1:
https://github.com/vmware-tanzu/velero/commit/9996dc5ce9506834c20fcbfba5787aa10301102b
https://github.com/vmware-tanzu/velero/commit/1da212b0e3cf4b74a711885e27495f6d302ed51c
https://github.com/vmware-tanzu/velero/commit/37a712ef2fd4f235a41dc5dd1116cbb84ffa6d9e
https://github.com/vmware-tanzu/velero/commit/8c6ed31528fe1c5ed81536a6b22d285e5bdbe266

I don't think any change would impact the code path for S3-compatible object storage.  
We also tested v1.7.1-rc1 in the pipeline and verified with AWS so the default BSL with defaul credentials should also work.

The bundle you provided for `test9` contains the patched BSL, and I didn't find any problem.

Before you see the error for test9, did you create the BSL manually?  It may be possible you faced the issue like https://github.com/vmware-tanzu/velero/issues/4196, that the BSL pointed to empty secret so the default env var will not be read.

**Comment by lautenbde** [comment]
@reasonerjt I turned out I was facing a glitch - In my case Velero is installed with options --no-default-backup-location --no-secret, default Secret and BSL are created post install. This will cause Velero Pod to restart in order to adopt the new mount and ENV variable for the secret. As my cluster was short on resources, the new Velero Pod was stuck in Pending state, so I was testing with the original Pod not having the secret mounted yet... I did not realize it when I was opening the ticket... I am sorry - thanks for your support!
